packages/seda-chain-contracts/src/data_request.rs

```diff
@@ -1,21 +1,29 @@
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::{Deps, DepsMut, MessageInfo, Order, Response, StdResult};
-use cw_storage_plus::Bound;
 
-use crate::msg::{GetDataRequestResponse, GetDataRequestsFromPoolResponse, PostDataRequestArgs};
+use crate::state::{DATA_REQUESTS, DATA_REQUESTS_COUNT};
+
+use crate::msg::{GetDataRequestResponse, GetDataRequestsFromPoolResponse};
 use crate::state::DataRequest;
-use crate::state::{DATA_REQUESTS_BY_NONCE, DATA_REQUESTS_COUNT, DATA_REQUESTS_POOL, DATA_RESULTS};
 use crate::types::Hash;
-use crate::utils::hash_data_request;
 use crate::ContractError;
 
 pub mod data_requests {
+    use std::collections::HashMap;
+
+    use crate::{
+        msg::PostDataRequestArgs,
+        state::{DATA_REQUESTS_BY_NONCE, DATA_RESULTS},
+    };
+    use cw_storage_plus::Bound;
+    use sha3::Digest;
+    use sha3::Keccak256;
 
     use super::*;
 
     /// Internal function to return whether a data request or result exists with the given id.
     fn data_request_or_result_exists(deps: Deps, dr_id: Hash) -> bool {
```

Comment:
I think this function can take a reference to a `Hash` for `dr_id` instead because I believe the `may_load` function could take a reference to `dr_id` instead of needing to clone it. But I'm not 100% sure. At leat based on the [definition ](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.Map.html#method.may_load) this seems true.

Reply:
Same here, unfortunately, it doesn't accept a reference.